command/command.go

```diff
@@ -552,24 +579,51 @@ func reportCheckMonitors(app *App, customIdentifier string, reports []*checks.Re
 		}
 	}
 	for {
-		err := app.API.ReportCheckMonitors(hostID, reports)
+		err := reportCheckMonitorsInternal(app, hostID, reports)
 		if err == nil {
 			break
 		}
-
-		logger.Errorf("ReportCheckMonitors: %s", err)
-
+		// give up on client error
 		if mackerel.IsClientError(err) {
 			break
 		}
 
-		logger.Debugf("ReportCheckMonitors: Sleep %d seconds before reporting", reportCheckRetryDelaySeconds)
+		logger.Debugf("ReportCheckMonitors: Sleep %d seconds before reporting again", reportCheckRetryDelaySeconds)
 
 		// retry until report succeeds
 		time.Sleep(time.Duration(reportCheckRetryDelaySeconds) * time.Second)
 	}
 }
 
+func reportCheckMonitorsInternal(app *App, hostID string, reports []*checks.Report) (err error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), 25*time.Second)
+	defer cancel()
+
+	err = app.API.ReportCheckMonitors(hostID, reports)
+	if err == nil {
+		logger.Debugf("ReportCheckMonitors: succeeded")
+		return
+	}
+	// Do not retry when first request already took long.
+	// (Typically when first request has timeout)
+	if ctxErr := ctx.Err(); ctxErr != nil {
+		logger.Warningf("ReportCheckMonitors: failed to report (will retry): %s", err)
```

Comment:
Should this use `ctxErr` instead of `err`?

Reply:
Here `ctx` is used just for handling timeout, so there's no useful information for logging.  So I'd like to log original err.
(So should I use just something like `start := time.Now(); ... if time.Now().After(start.Add25 * time.Second)) { ... }` to avoid confusion?)

Reply:
Okay, I rewrote not to use `ctx`